bittensor/commands/stake.py

```diff
@@ -610,81 +895,94 @@ def run(cli: "bittensor.cli"):
 
     @staticmethod
     def _run(cli: "bittensor.cli", subtensor: "bittensor.subtensor"):
+        console = Console()
         wallet = bittensor.wallet(config=cli.config)
 
         # Get values if not set.
-        if not cli.config.is_set("netuid"):
-            cli.config.netuid = int(Prompt.ask("Enter netuid"))
-
-        netuid = cli.config.netuid
-        total_subnets = subtensor.get_total_subnets()
-        if total_subnets is not None and total_subnets <= netuid <= 0:
-            raise ValueError("Netuid is outside the current subnet range")
+        exists, netuid = get_netuid(cli, subtensor)
+        if not exists:
+            return
 
-        if not cli.config.is_set("hotkey"):
-            cli.config.hotkey = Prompt.ask("Enter parent hotkey (ss58)")
-        if not wallet_utils.is_valid_ss58_address(cli.config.hotkey):
-            console.print(
-                f":cross_mark:[red] Invalid SS58 address: {cli.config.hotkey}[/red]"
-            )
+        # get parent hotkey
+        hotkey = get_hotkey(wallet, cli.config)
+        if not wallet_utils.is_valid_ss58_address(hotkey):
+            console.print(f":cross_mark:[red] Invalid SS58 address: {hotkey}[/red]")
             return
 
-        # get children
+        # get current children
         curr_children = GetChildrenCommand.retrieve_children(
             subtensor=subtensor,
-            hotkey=cli.config.hotkey,
-            netuid=cli.config.netuid,
+            hotkey=hotkey,
+            netuid=netuid,
             render_table=False,
         )
 
         if curr_children:
-            GetChildrenCommand.retrieve_children(
+            # print the table of current children
+            hotkey_stake = subtensor.get_total_stake_for_hotkey(hotkey)
+            GetChildrenCommand.render_table(
                 subtensor=subtensor,
-                hotkey=cli.config.hotkey,
-                netuid=cli.config.netuid,
-                render_table=True,
-            )
-            raise ValueError(
-                f"There are already children hotkeys under parent hotkey {cli.config.hotkey}. "
-                f"Call revoke_children command before attempting to set_children again, or call the get_children command to view them."
+                hotkey=hotkey,
+                hotkey_stake=hotkey_stake,
+                children=curr_children,
+                netuid=netuid,
+                prompt=False,
             )
 
+        # get new children
         if not cli.config.is_set("children"):
             cli.config.children = Prompt.ask(
-                "Enter child(ren) hotkeys (ss58) as comma-separated values"
+                "Enter child hotkeys (ss58) as comma-separated values"
             )
-        children = [str(x) for x in re.split(r"[ ,]+", cli.config.children)]
+        proposed_children = [str(x) for x in re.split(r"[ ,]+", cli.config.children)]
+
+        # Set max 5 children
+        if len(proposed_children) > 5:
```

Comment:
maybe it's worth keeping this (`5`) as a constant at the top of the module to be more obvious about this limit?